core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1304,6 +1383,221 @@ static void handle_mem_reclaim(struct thread_smc_args *args)
 out:
 	spmc_set_args(args, ret_fid, ret_val, 0, 0, 0, 0);
 }
+
+static void handle_notification_bitmap_create(struct thread_smc_args *args)
+{
+	uint32_t ret_val = FFA_INVALID_PARAMETERS;
+	uint32_t ret_fid = FFA_ERROR;
+	uint32_t old_itr_status = 0;
+
+	if (!FFA_TARGET_INFO_GET_SP_ID(args->a1) && !args->a3 && !args->a4 &&
+	    !args->a5 && !args->a6 && !args->a7) {
+		uint16_t vm_id = args->a1;
+
+		old_itr_status = cpu_spin_lock_xsave(&spmc_notif_lock);
+
+		if (notif_vm_id_valid) {
+			if (vm_id == notif_vm_id)
+				ret_val = FFA_DENIED;
+			else
+				ret_val = FFA_NO_MEMORY;
+		} else {
+			notif_vm_id = vm_id;
+			notif_vm_id_valid = true;
+			ret_val = FFA_OK;
+			ret_fid = FFA_SUCCESS_32;
+		}
+
+		cpu_spin_unlock_xrestore(&spmc_notif_lock, old_itr_status);
+	}
+
+	spmc_set_args(args, ret_fid, 0, ret_val, 0, 0, 0);
+}
+
+static void handle_notification_bitmap_destroy(struct thread_smc_args *args)
+{
+	uint32_t ret_val = FFA_INVALID_PARAMETERS;
+	uint32_t ret_fid = FFA_ERROR;
+	uint32_t old_itr_status = 0;
+
+	if (!FFA_TARGET_INFO_GET_SP_ID(args->a1) && !args->a3 && !args->a4 &&
+	    !args->a5 && !args->a6 && !args->a7) {
+		uint16_t vm_id = args->a1;
+
+		old_itr_status = cpu_spin_lock_xsave(&spmc_notif_lock);
+
+		if (!notif_async_is_started() && notif_vm_id_valid &&
```

Comment:
From the FF-A spec, return code should be DENIED "if the Notification bitmap is not registered or is
registered but not in a masked and non-pending state.".

Reply:
Thanks, I'll update.